File: 392-is-subsequence/is-subsequence.ts
function isSubsequence(s: string, t: string): boolean {
    if (s.length > t.length){
        return false;
    }
    let p1 = 0
    let p2 = 0
    while (p1<s.length && p2<t.length){
        if (s[p1]==t[p2]){
            p1++
            if (s.length ==p1){
                return true
            }
        }
        p2++;
    }
    return p1 == s.length
};

File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s)>len(t): return False

        p1,p2 = 0,0
        while(p2<len(t) and p1 <len(s)):
            if t[p2] == s[p1]:
                p1+=1
                if p1 == len(s):
                    return True
            p2+=1
        return p1 == len(s)